griptape/drivers/prompt/huggingface_pipeline_prompt_driver.py

```diff
@@ -35,23 +36,24 @@ class HuggingFacePipelinePromptDriver(BasePromptDriver):
         ),
         kw_only=True,
     )
-    pipe: TextGenerationPipeline = field(
-        default=Factory(
-            lambda self: import_optional_dependency("transformers").pipeline(
-                "text-generation",
-                model=self.model,
-                max_new_tokens=self.max_tokens,
-                tokenizer=self.tokenizer.tokenizer,
-            ),
-            takes_self=True,
-        ),
+    _text_generation_pipeline: TextGenerationPipeline = field(
+        default=None, kw_only=True, alias="text_generation_pipeline", metadata={"serializable": False}
     )
```

Comment:
I know I'm contradicting a point I made offline, but what if we change this to support non-text generation pipelines in the future? Maybe it's better to name off the base class: `Pipeline`.